### PR TITLE
Unit testcase

### PR DESCRIPTION
**Pull request by KajalMathur**
Kindly review my PR.

### CODE REVIEW

File: src/main/java/com/product/dao/ProductDaoImpl.java [hunk 1]
```diff
@@ -9,6 +9,9 @@
 
 import com.product.model.Product;
 
+import lombok.AllArgsConstructor;
```
Comment (AakashIsTheLimit) [inline src/main/java/com/product/dao/ProductDaoImpl.java]: Unnecessary inports

File: src/main/java/com/product/model/Address.java [hunk 2]
```diff
@@ -1,8 +1,16 @@
 package com.product.model;
 
+import java.util.List;
```
Comment (AakashIsTheLimit) [inline src/main/java/com/product/model/Address.java]: Quite many unnecessary imports

File: src/main/java/com/product/service/ProductService.java [hunk 3]
```diff
@@ -8,11 +8,11 @@
 
 public interface ProductService {
 
-	public void deleteProductById(int id);
+	public void deleteProductbyId(int id);
 
 	public void updateProduct(Product emp);
 
-	public void deleteAllProducts();
+	public void deleteallProducts();
```
Comment (AakashIsTheLimit) [inline src/main/java/com/product/service/ProductService.java]: It should be the previous one.. deleteAll....

File: src/test/java/com/product/productService/ProductServiceApplicationTest.java [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+
+		/* Then */
+		Assert.assertEquals(productNewList, productList);
+	}
+
+	@Test
+	public void getAllProductDescTest() {
+
+		/* Given */
+	    productList = new ArrayList<Product>();
```
Comment (AakashIsTheLimit) [inline src/test/java/com/product/productService/ProductServiceApplicationTest.java:94]: Formatting seems off

File: src/test/java/com/product/productService/ProductServiceApplicationTest.java [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+	private List<Product> productList;
+	private  Product product1;
+	private  Product product2;
+
+	@Before
+	public void preRequisiteDataCreation() {
+
+		/* Data Creation */
+		ImmutableList<String> othersize = ImmutableList.of("Rick", "Ron", "Victor");
+		Map<String, Integer> details = new LinkedHashMap<String, Integer>();
```
Comment (AakashIsTheLimit) [inline src/test/java/com/product/productService/ProductServiceApplicationTest.java:48]: Does it need to maintain the order ??

File: src/test/java/com/product/productService/CustomerProductServiceApplicationTest.java [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+		/* When */
+		CustomerProductResponse customerProductResponse = customerProductServiceImpl.getcustomerProductById(1, 1);
+
+		/* Then */
+		Assert.assertEquals(customerProductResponse.getProduct().getId(), 1);
+		Assert.assertEquals(customerProductResponse.getCustomerResponse().getFirstName(), "Ram");
+	}
+
+	@Test
+	public void verifyFindCustomerProductResponseIfProductNotPresent() {
```
Comment (AakashIsTheLimit) [inline src/test/java/com/product/productService/CustomerProductServiceApplicationTest.java:105]: What happens if both are not present ..
Missing test case

File: src/main/java/com/product/dao/ProductDaoImpl.java (hunk 1 above)
Comment (KajalMathur) [reply src/main/java/com/product/dao/ProductDaoImpl.java]: Done.

File: src/main/java/com/product/model/Address.java (hunk 2 above)
Comment (KajalMathur) [reply src/main/java/com/product/model/Address.java]: Done.

File: src/main/java/com/product/service/ProductService.java (hunk 3 above)
Comment (KajalMathur) [reply src/main/java/com/product/service/ProductService.java]: Done.

File: src/test/java/com/product/productService/ProductServiceApplicationTest.java (hunk 5 above)
Comment (KajalMathur) [reply src/test/java/com/product/productService/ProductServiceApplicationTest.java:48]: yes.

File: src/main/java/com/product/model/Manufacturer.java [hunk 7]
```diff
@@ -1,15 +1,25 @@
 package com.product.model;
 
+import java.util.Date;
```
Comment (AakashIsTheLimit) [inline src/main/java/com/product/model/Manufacturer.java:3]: unnecessary imports

File: src/main/java/com/product/model/Product.java [hunk 8]
```diff
@@ -1,17 +1,22 @@
 package com.product.model;
 
+import java.util.Date;
```
Comment (AakashIsTheLimit) [inline src/main/java/com/product/model/Product.java:3]: unnecessary imports

please enable setting on your IDE to get rid of them automatically (or when you format the code)

File: src/main/java/com/product/service/CustomerProductService.java [hunk 9]
```diff
@@ -4,5 +4,5 @@
 
 public interface CustomerProductService {
 
-	public CustomerProductResponse getcustometerProductById(int customerId, int productId);
+	public CustomerProductResponse getcustomerProductById(int customerId, int productId);
```
Comment (AakashIsTheLimit) [inline src/main/java/com/product/service/CustomerProductService.java:7]: change this to `getCust..`

File: src/test/java/com/product/productService/CustomerProductServiceApplicationTest.java (hunk 6 above)
Comment (AakashIsTheLimit) [reply src/test/java/com/product/productService/CustomerProductServiceApplicationTest.java:105]: you did not cover thhis case ^

File: src/test/java/com/product/productService/ProductServiceApplicationTest.java (hunk 5 above)
Comment (AakashIsTheLimit) [reply src/test/java/com/product/productService/ProductServiceApplicationTest.java:48]: why ? this is details .. who cares ?

I saw u used `mmutableMap<String, Integer> details = ImmutableMap.of("quantity", 1, "sold",2);` in the other test class. Why not in this one ?

File: src/test/java/com/product/productService/ProductServiceApplicationTest.java (hunk 4 above)
Comment (AakashIsTheLimit) [reply src/test/java/com/product/productService/ProductServiceApplicationTest.java:94]: formatting is still off

**Pull request closed without merging**